src/script/cpp_api/s_server.cpp

```diff
@@ -196,3 +197,68 @@ std::string ScriptApiServer::formatChatMessage(const std::string &name,
 
 	return ret;
 }
+
+u32 ScriptApiServer::allocateDynamicMediaCallback(int f_idx)
+{
+	lua_State *L = getStack();
+
+	if (f_idx < 0)
+		f_idx = lua_gettop(L) + f_idx + 1;
+
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "dynamic_media_callbacks");
+	luaL_checktype(L, -1, LUA_TTABLE);
+
+	// Find a randomly generated token that doesn't exist yet
+	int tries = 100;
+	u32 token;
+	while (1) {
```

Comment:
That makes the exit logic a bit more awkward:
```cpp
	for (int tries = 0; tries < 100; ++tries) {
		// ...
		if (tries == 99)
			FATAL_ERROR("Ran out of callbacks IDs?!");
	}
```

Reply:
`goto` inside the loop like this would be possible too, but it's fighting poison with poison and one is worse than the other :confused: 

```C++
for (int tries = 0; tries < 100; ++tries) {
	// ...
	if (is_free)
		goto horray_i_am_free;
}
FATAL_ERROR("Ran out of callbacks IDs?!");

horray_i_am_free:
```